exopy_qm/tasks/tasks/ConfigureExecuteTask.py

```diff
@@ -0,0 +1,264 @@
+
+    #: Parameters entered by the user for the program and config
+    parameters = Typed(dict).tag(pref=True)
+
+    #: Comments associated with the parameters
+    comments = Typed(dict).tag(pref=True)
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._config_module = None
+        self._program_module = None
+        self.parameters = {}
+        self.comments = {}
+
+    def check(self, *args, **kwargs):
+        test, traceback = super(ConfigureExecuteTask,
+                                self).check(*args, **kwargs)
+
+        if not test:
+            return test, traceback
+
+        if self._config_module is None or self._program_module is None:
+            msg = ('Config or program missing')
+            traceback[self.get_error_path() + '-trace'] = msg
+
+        for key, value in self.parameters.items():
+            try:
+                self.format_and_eval_string(value)
+            except:
+                e = sys.exc_info()[0]
+                msg = ("Couldn't evaluate {} : {}")
+                traceback[self.get_error_path() + '-trace'] = msg.format(value, e)
+
+        return test, traceback
+
+    def perform(self):
+        self._update_parameters()
+
+        # Evaluate all parameters
+        evaluated_parameters = {}
+        for key, value in self.parameters.items():
+            evaluated_parameters[key] = self.format_and_eval_string(value)
+
+        config_to_set = self._config_module.get_config(evaluated_parameters)
+        program_to_execute = self._program_module.get_prog(
+            evaluated_parameters)
+
+        self.driver.set_config(config_to_set)
+        self.driver.execute_program(
+            program_to_execute, self.duration_limit, self.data_limit)
+        time.sleep(self.wait_time)
+        results = self.driver.get_results()
+
+        for k in results.variable_results.__dict__:
+            self.write_in_database(
+                'variable_' + k, getattr(results.variable_results, k).data)
+            if getattr(results.variable_results, k).possible_data_loss:
+                log = logging.getLogger()
+                log.warning(
+                    f"[Variable {k}] Possible data loss detected, you should increase the waiting time")
+
+        for k in results.raw_results.__dict__:
+            self.write_in_database(
+                'raw_' + k + '_1', getattr(results.raw_results, k).input1_data)
+            self.write_in_database(
+                'raw_' + k + '_2', getattr(results.raw_results, k).input2_data)
+
+            # All the values in the data_loss array are identical
+            if getattr(results.raw_results, k).data_loss[0]:
+                log = logging.getLogger()
+                log.warning(
+                    f"[Trace {k}] Data loss detected, you should increase the waiting time")
+
+    def _post_setattr_path_to_program_file(self, old, new):
+        if new or new != '':
+            importlib.invalidate_caches()
```

Comment:
Hopefully this won't bite you ! I might be nicer to simply remove the cached value you care about (loading possibly two different numpy sounds like a bad idea).

Reply:
I don't think I have a choice. If I don't invalidate the caches, changes made to the configuration or program file might not be taken into account. 
I agree that if a library is update while exopy is running, "funny" things could happen.

Reply:
Just read the docs of `invalidate_cache` and it is not as bad as I though. So we can keep it this way.